x-pack/plugins/cloud_integrations/cloud_experiments/public/launch_darkly_client/launch_darkly_client.ts

```diff
@@ -26,7 +27,8 @@ export interface LaunchDarklyUserMetadata
 }
 
 export class LaunchDarklyClient {
-  private launchDarklyClient?: LDClient;
+  private launchDarklyClient?: Promise<LDClient | null>;
```

Comment:
Good idea! I'd give this direction a try. 
Not sure if I will pursue finalizing this PR though since I was unblocked by using server-side client.

First my direction made sense when I thought that the problem only in the lazy loading of the client, but then it got more complicated when I realized that `updateUserMetadata` can be called after the first `getVariation` call

Reply:
Yeah... makes sense! Let's keep this PR as is for now... hopefully any of us (or someone else) can come back to this and finalize it :)